core/src/main/java/com/orange/ouds/core/component/OudsSwitch.kt

```diff
@@ -0,0 +1,283 @@
+                    tint = checkColor
+                )
+            }
+        }
+
+        Spacer(modifier = Modifier.weight(1.0f))
+    }
+}
+
+@Composable
+private fun indicatorBackgroundColor(state: OudsControl.State, checked: Boolean): Color {
+    return with(OudsTheme.componentsTokens.switch) {
+        when (state) {
+            OudsControl.State.Enabled -> if (checked) colorTrackSelected.value else colorTrackUnselected.value
+            OudsControl.State.Hovered,
+            OudsControl.State.Pressed,
+            OudsControl.State.Focused -> if (checked) colorTrackSelectedInteraction.value else colorTrackUnselectedInteraction.value
+            OudsControl.State.Disabled -> OudsTheme.colorScheme.action.disabled
+        }
+    }
+}
+
+@Composable
+private fun cursorSize(state: OudsControl.State, checked: Boolean): Size {
+    return with(OudsTheme.componentsTokens.switch) {
+        val width = when {
+            state == OudsControl.State.Pressed && checked -> sizeWidthCursorSelectedPressed
+            state == OudsControl.State.Pressed && !checked -> sizeWidthCursorUnselectedPressed
+            checked -> sizeWidthCursorSelected
+            else -> sizeWidthCursorUnselected
+        }
+        val height = if (checked) sizeHeightCursorSelected else sizeHeightCursorUnselected
+        Size(width, height)
+    }
+}
+
+@Composable
+private fun checkColor(state: OudsControl.State, checked: Boolean): Color? {
+    return if (checked) {
+        when (state) {
+            OudsControl.State.Enabled,
+            OudsControl.State.Hovered,
+            OudsControl.State.Focused -> OudsTheme.componentsTokens.switch.colorCheck.value
+            OudsControl.State.Pressed -> null
+            OudsControl.State.Disabled -> OudsTheme.colorScheme.action.disabled
+        }
+    } else {
+        null
+    }
+}
+
+private fun <T> cursorAnimationSpec(): AnimationSpec<T> {
+    return tween(
+        durationMillis = OudsSwitch.AnimationDuration,
+        easing = CubicBezierEasing(0.2f, 0.0f, 0.0f, 1.0f)
+    )
+}
+
+@Composable
+private fun animateHorizontalAlignmentAsState(targetBiasValue: Float, animationSpec: AnimationSpec<Float> = spring()): State<BiasAlignment.Horizontal> {
+    val bias by animateFloatAsState(targetBiasValue, animationSpec)
+    return remember { derivedStateOf { BiasAlignment.Horizontal(bias) } }
+}
+
+private object OudsSwitch {
+
+    val AnimationDuration = 150
+}
+
+@UiModePreviews.Default
+@Composable
+@Suppress("PreviewShouldNotBeCalledRecursively")
+private fun PreviewOudsSwitch(@PreviewParameter(OudsSwitchPreviewParameterProvider::class) checked: Boolean) {
+    PreviewOudsSwitch(darkThemeEnabled = isSystemInDarkTheme(), checked = checked)
+}
+
+@Composable
+internal fun PreviewOudsSwitch(
+    darkThemeEnabled: Boolean,
+    checked: Boolean
+) = OudsPreview(darkThemeEnabled = darkThemeEnabled) {
+    with(checked) {
+        val columnCount = 2
+        Box(modifier = Modifier.padding(16.dp)) {
```

Comment:
We are using the `16.dp` value in all previews. Maybe we could factorize it as a constant in the `OudsPreview` companion object?

Reply:
This code is factorized in #464, but maybe we should also factorize the three occurences of `16.dp` there.

Reply:
No problem, we can do that in the other PR